frontend/src/utils/session/useCurrentBrowserSessionId.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { CombinedError } from "@urql/core";
+import { useAtomValue } from "jotai";
+
+import { currentBrowserSessionIdAtom } from "../../atoms";
+import { isOk, unwrapOk, unwrapErr, isErr } from "../../result";
+
+/**
+ * Query the current browser session id
+ * and unwrap the result
+ */
+export const useCurrentBrowserSessionId = (): {
```

Comment:
I don't understand the value of this, especially looking at the usage? Is it for easier mocking? If that's the case, you can instead inject the atom value with [`useHydrateAtoms`](https://jotai.org/docs/utilities/ssr)

Reply:
The idea is to avoid having to do the error checking/handling and unwrapping in components. After adding the [error boundary](https://github.com/matrix-org/matrix-authentication-service/pull/1743) I simplify this to throw on error, so consuming components don't have to check the error and render the error page. 
https://github.com/matrix-org/matrix-authentication-service/pull/1743/files#diff-0c5b8a4159d243e8dd602e3530f1e41e9e6ec98cc0514317fd7abda2350f5369

Reply:
Alright, makes sense